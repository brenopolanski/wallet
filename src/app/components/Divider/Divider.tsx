import { styled } from "twin.macro";

import { getStyles } from "./Divider.styles";

interface DividerProperties {
	type?: "horizontal" | "vertical";
	size?: "sm" | "md" | "lg" | "xl";
	dashed?: boolean;
	className?: string;
}

export const Divider = styled.div.attrs<DividerProperties>(({ className, type }) => ({
	className: className ? undefined : "border-theme-secondary-300 dark:border-theme-secondary-800",
	type: type ?? "horizontal",
}))<DividerProperties>(getStyles);
