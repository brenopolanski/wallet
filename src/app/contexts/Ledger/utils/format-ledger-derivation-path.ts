import { LedgerDerivationScheme } from "app/contexts/Ledger/contracts";

export const formatLedgerDerivationPath = (scheme: LedgerDerivationScheme) =>
	`m/${scheme.purpose || 44}'/${scheme.coinType}'/${scheme.account || 0}'/${scheme.change || 0}/${
		scheme.address || 0
	}`;
