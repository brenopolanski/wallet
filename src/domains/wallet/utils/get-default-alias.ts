import { Networks } from "@payvo/sdk";
import { Contracts } from "@payvo/sdk-profiles";

interface GetDefaultAliasInput {
	profile: Contracts.IProfile;
	network: Networks.Network;
}

export const getDefaultAlias = ({ profile, network }: GetDefaultAliasInput): string => {
	const makeAlias = (count: number) => `${network.displayName()} #${count}`;

	const sameCoinWallets = profile.wallets().findByCoinWithNetwork(network.coin(), network.id());

	let sameCoinWalletsCount = sameCoinWallets.length || 1;

	while (profile.wallets().findByAlias(makeAlias(sameCoinWalletsCount))) {
		sameCoinWalletsCount++;
	}

	return makeAlias(sameCoinWalletsCount);
};
