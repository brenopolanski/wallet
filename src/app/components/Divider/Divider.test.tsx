import React from "react";
import { render } from "utils/testing-library";

import { Divider } from "./Divider";

describe("Divider", () => {
	const sizes = ["sm", "md", "lg", "default"];

	it("should render", () => {
		const { container, asFragment } = render(<Divider />);

		expect(container).toBeInTheDocument();
		expect(asFragment()).toMatchSnapshot();
	});

	it("should render horizontal type", () => {
		const { container } = render(<Divider type="horizontal" />);

		expect(container).toMatchSnapshot();
	});

	it("should render vertical type", () => {
		const { container } = render(<Divider type="vertical" />);

		expect(container).toMatchSnapshot();
	});

	it.each(sizes)("should render vertical with size %s", (size) => {
		const { container } = render(<Divider type="vertical" size={size as any} />);

		expect(container).toMatchSnapshot();
	});

	it.each(sizes)("should render horizontal with size %s", (size) => {
		const { container } = render(<Divider type="horizontal" size={size as any} />);

		expect(container).toMatchSnapshot();
	});

	it("should render a dashed", () => {
		const { container } = render(<Divider dashed />);

		expect(container).toMatchSnapshot();
	});
});
