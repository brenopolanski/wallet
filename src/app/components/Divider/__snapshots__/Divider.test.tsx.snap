// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider should render 1`] = `
<DocumentFragment>
  <div
    class="Divider-sc-17ehhxk-0 hoeInf border-theme-secondary-300 dark:border-theme-secondary-800"
    type="horizontal"
  />
</DocumentFragment>
`;

exports[`Divider should render a dashed 1`] = `
<div>
  <div
    class="Divider-sc-17ehhxk-0 FRyyK border-theme-secondary-300 dark:border-theme-secondary-800"
    type="horizontal"
  />
</div>
`;

exports[`Divider should render horizontal type 1`] = `
<div>
  <div
    class="Divider-sc-17ehhxk-0 hoeInf border-theme-secondary-300 dark:border-theme-secondary-800"
    type="horizontal"
  />
</div>
`;

exports[`Divider should render horizontal with size default 1`] = `
<div>
  <div
    class="Divider-sc-17ehhxk-0 hoeInf border-theme-secondary-300 dark:border-theme-secondary-800"
    type="horizontal"
  />
</div>
`;

exports[`Divider should render horizontal with size lg 1`] = `
<div>
  <div
    class="Divider-sc-17ehhxk-0 hoeInf border-theme-secondary-300 dark:border-theme-secondary-800"
    type="horizontal"
  />
</div>
`;

exports[`Divider should render horizontal with size md 1`] = `
<div>
  <div
    class="Divider-sc-17ehhxk-0 hoeInf border-theme-secondary-300 dark:border-theme-secondary-800"
    type="horizontal"
  />
</div>
`;

exports[`Divider should render horizontal with size sm 1`] = `
<div>
  <div
    class="Divider-sc-17ehhxk-0 hoeInf border-theme-secondary-300 dark:border-theme-secondary-800"
    type="horizontal"
  />
</div>
`;

exports[`Divider should render vertical type 1`] = `
<div>
  <div
    class="Divider-sc-17ehhxk-0 jqFYpY border-theme-secondary-300 dark:border-theme-secondary-800"
    type="vertical"
  />
</div>
`;

exports[`Divider should render vertical with size default 1`] = `
<div>
  <div
    class="Divider-sc-17ehhxk-0 jqFYpY border-theme-secondary-300 dark:border-theme-secondary-800"
    type="vertical"
  />
</div>
`;

exports[`Divider should render vertical with size lg 1`] = `
<div>
  <div
    class="Divider-sc-17ehhxk-0 bGhmfz border-theme-secondary-300 dark:border-theme-secondary-800"
    type="vertical"
  />
</div>
`;

exports[`Divider should render vertical with size md 1`] = `
<div>
  <div
    class="Divider-sc-17ehhxk-0 hhDSpl border-theme-secondary-300 dark:border-theme-secondary-800"
    type="vertical"
  />
</div>
`;

exports[`Divider should render vertical with size sm 1`] = `
<div>
  <div
    class="Divider-sc-17ehhxk-0 jOzFza border-theme-secondary-300 dark:border-theme-secondary-800"
    type="vertical"
  />
</div>
`;
